### ISSUE TITLE
Only request unlocking of database for websites for which there is a record

### ISSUE THREAD

**Issue by abustany** [smorks/keepasshttp-connector#78]
Apologies if this question has already been asked (I tried to look in old issues first), but would it be possible to somehow keep a cache of all known URLs (those would of course be unprotected in memory), and only ask to unlock the database when a known website is detected? Right now, it seems that my database has to be permanently unlocked for the connector to operate properly.

**Comment by smorks** [comment]
yes, correct, the database has to be unlocked for the extension to work.

i'm guessing with what you are suggesting it would work something like this?

1. Browser is opened, if database is unlocked it sends all possible url's to the extension, which it stores.
2. Database is locked.
3. Continue browsing as normal, if it hits a login page for a know url, it prompts to unlock the database
4. Credentials are sent to extension
5. Database is re-locked.

**Comment by abustany** [comment]
yes, you got it right. I don't think we even need to cache the list of URLs anywhere else than in memory... One question being, can the extension "listen" to entries being updated/added/removed from the DB, so that the list of URLs in memory can be kept up to date...

**Comment by smorks** [comment]
it's not currently possible for this extension to listen, but may be possible with this one: https://github.com/varjolintu/keepassxc-browser, it would  just be a matter of implementing it.

**Comment by abustany** [comment]
in other words, the "listening" feature is only possible using native messaging, and not over the HTTP interface ?

**Comment by smorks** [comment]
correct. i mean it might be possible to use WebSockets or something, but i think moving to native messaging is a more secure way to go anyways.
